src/main/java/duke/components/Parser.java

```diff
@@ -0,0 +1,227 @@
+package duke.components;
+
+import java.time.LocalDateTime;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+import duke.task.Deadline;
+import duke.task.Event;
+import duke.task.ToDos;
+
+/**
+ * Represents the logic used to handle user inputs, and parse user
+ * input as commands to the chatBot.
+ */
+public class Parser {
+    private Ui ui;
+    private FileStorage fileStorage;
+    private TaskList fullList;
+    private DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm");
+
+    /**
+     * Constructs a Parser instance with the given file path.
+     *
+     * @param filepath The path to the file used for storing tasks.
+     */
+    public Parser(String filepath) {
+        this.ui = new Ui();
+        this.fileStorage = new FileStorage(filepath);
+        this.fullList = new TaskList();
+    }
+
+    /**
+     * Constructs a Parser instance with the specified parameters.
+     *
+     * @param fileStorage The FileStorage object to handle file operations.
+     * @param fullList The TaskList object to store ttasks.
+     * @param ui The Ui object to display outputs to the user.
+     */
+    public Parser(FileStorage fileStorage, TaskList fullList, Ui ui) {
+        this.ui = ui;
+        this.fileStorage = fileStorage;
+        this.fullList = fullList;
+    }
+
+    /**
+     * Represents the various command types that can be used in the chatBot application.
+     * These command types define the actions the chatBot can perform.
+     */
+    public enum CommandType {
+        TODO, DEADLINE, EVENT, MARK, UNMARK,
+        DELETE, LIST, BYE, UNKNOWN, EMPTY, FIND
+    }
+
+    /**
+     * Returns a CommandType based on the user input.
+     *
+     * @param input User input to be parsed.
+     * @return The CommandType enum representing the parsed user input.
+     */
+    public CommandType parseCommand(String input) {
+        if (input.equals("bye")) {
+            return CommandType.BYE;
+        } else if (input.equals("mark")) {
+            return CommandType.MARK;
+        } else if (input.equals("unmark")) {
+            return CommandType.UNMARK;
+        } else if (input.equals("todo")) {
+            return CommandType.TODO;
+        } else if (input.equals("deadline")) {
+            return CommandType.DEADLINE;
+        } else if (input.equals("event")) {
+            return CommandType.EVENT;
+        } else if (input.equals("delete")) {
+            return CommandType.DELETE;
+        } else if (input.equals("list")) {
+            return CommandType.LIST;
+        } else if (input.equals("")) {
+            return CommandType.EMPTY;
+        } else if (input.equals("find")) {
+            return CommandType.FIND;
+        } else {
+            return CommandType.UNKNOWN;
+        }
```

Comment:
This if statement is very long -- would it be more readable if it was a switch-case instead?
You can consider using something like [`CommandType.valueOf()`](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/lang/Enum.html#valueOf(java.lang.Class,java.lang.String)) to make the code more concise.